ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestinationSync.java

```diff
@@ -330,6 +328,10 @@ protected BlobProperties getBlobProperties(AzureBlobLayoutStrategy.BlobLayout bl
     try {
       return createOrGetBlobStore(blobLayout.containerName).getBlobClient(blobLayout.blobFilePath).getProperties();
     } catch (BlobStorageException bse) {
+      if (bse.getErrorCode() == BlobErrorCode.BLOB_NOT_FOUND) {
```

Comment:
I suggest we still throw BLOB_NOT_FOUND, just let getBlobMetadata handle it in the right way.

Reply:
done